### ISSUE TITLE
Migrate to GitLab?

### ISSUE THREAD

**Issue by Zughy** [LibreSprite/LibreSprite#161]
One thing that usually annoys me, is how the vast majority of free software projects are on a git platform (GitHub) that lies on a proprietary JS core. A platform that belongs to a company that embraces "open source" basically just for marketing, the same one that called Linux (I quote) "a cancer" in 2001, requiring [19 years](https://www.theverge.com/2020/5/18/21262103/microsoft-open-source-linux-history-wrong-statement) to take that back and that tries to make people addict to its walled gardens (see Minecraft accounts merged with Microsoft ones last year). For a further reading about Microsoft critiques, see https://en.wikipedia.org/wiki/Criticism_of_Microsoft, but in short, I can state hands down that Microsoft can't be trusted because you don't become a big tech doing good actions (another depressing example is Bezos' wealth showed in this site https://mkorostoff.github.io/1-pixel-wealth/).

I personally think that prevention is better than cure, so...: how about moving the project onto a git platform that doesn't belong to any big oligopoly and, most important, that is actually free software itself? This allows us to:
- give the good example (a lot of people don't move onto other services because they think there is no one there)
- create our own GitLab instance in case site ethics go downhill (e.g. they get bought by a big tech)

Also, in order to avoid disappearing, we could keep this repo as a mirror, so that people following the project from GitHub can still do that (till they notice it moved). And the same goes for the ones discovering it from GitHub, because it will still appear in search engines and in the internal GitHub search.

The more the time passes, the harder will be to make such migration (more issues, CI, website), especially because I've seen the fork number increasing in these days. So, yeah, I wanted to at least try to suggest it and see what other people think. I also add that I've been using both GitHub and Lab for a year now (browser and CLI) and I haven't noticed any big differences.

We're basically making good actions thanks to a questionable company that centralises its power: not a great perspective

**Comment by jjhaggar** [comment]
I think it would be nice having both (GitLab and GitHub) at the same time. That would act as a safety backup, while conserving the benefits of new users from GitHub.

About migrating completelly, I understand your concerns about Microsoft, but moving out of GitHub is probably not a good idea right now if we want LibreSprite to continue growing. Almost everyone who knows how to use Git a bit has a GitHub account (even not so very good developers, like me XD), and more users means that we can get help from more people. 

Every little barrier we put for potential contributors, like having to register on a new website, can take away people who would have helped if they could.

**Comment by Zughy** [comment]
> I think it would be nice having both (GitLab and GitHub) at the same time. That would act as a safety backup, while conserving the benefits of new users from GitHub.

So GitLab as a safe backup, right? I mean, with everything (PRs, issues) still happening on GitHub

> but moving out of GitHub is probably not a good idea right now if we want LibreSprite to continue growing.

I see your point, but there won't ever be a right moment. Now we need contributors, tomorrow we'll have too many things to move (issues, PRs, a better site relying on GitHub etc), there will always be a reason to remain here in the end.

Don't take this wrong, I don't want to insist and push my view, just trying to reason together: PerryHugh left, MintStudios is not answering, Calinou and Akien-mga are obviously busy working on Godot (where I think they're also paid, I'm genuinely happy for them), soo... if the project started breathing again, is mostly thanks to Maujin (mid 2020), me (second quarter 2021), and you (since the fork?). And now Felipe too. Yes, there are new forks, but nobody is currently working on anything (see here https://github.com/LibreSprite/LibreSprite/network). Also, for years LibreSprite was [basically dead](https://github.com/LibreSprite/LibreSprite/commits/master?after=3ea47a42044491e52335a72645a10938f4ba7351+34&branch=master), with just a few commits here and there about rebranding and removing the news section. In five years [almost nobody contributed](https://github.com/LibreSprite/LibreSprite/pulls?page=1&q=is%3Apr+is%3Aclosed) if we exclude maintainers. And we're talking about a project that had been forked from a very popular one (Aseprite), but even so, it was pretty tabula rasa after the first month of people angry at Aseprite for changing the license.

Now, I don't know how Maujin discovered LibreSprite and why he decided to contribute, but my experience wasn't through GitHub: I typed in "aseprite" in my distro's package manager and tadaa, here you have LibreSprite. And, as I care a lot about a project like this, no matter if it were on GitHub, GitLab, Notabug, Gitea etc., I would have decided to contribute anyway (creating an account if I had to). All of this to say that, yes, casual contributors are important, but they're not the ones actually bringing some change, that "ponen manos a la obra" to do something. You want motivated people, people that surely won't mind creating an account to improve something. Also, I don't have any statistics, but I'm pretty sure that LibreSprite users are mostly free software enthusiasts (because, being real, Aseprite has more features as for now). People like me that are probably happier to contribute to something on a platform that is not proprietary and that doesn't belong to some very questionable company which is all about capitalism and crushing the other.

Last but not least, I like to think LibreSprite is getting some attention because I'm spamming it in a few communities I'm in. So again, not because of an internal search, but because an active contributor tries to make the software more well-known.

In short: yes, we have more potential people here, but I think that there we'll have more motivated people, happy to know another project left GitHub

---

Also tagging @felipemanga and @maujin111 to know how they've discovered LibreSprite

**Comment by jjhaggar** [comment]
> So GitLab as a safe backup, right? I mean, with everything (PRs, issues) still happening on GitHub
That was my first idea, yeah, but maybe we could use GitLab as the primary one and have GitHub as a backup mirror. 

Yeah, that's right. What make projects grow more is indeed motivated people, so if some people can get motivation from the project going to GitLab, that's good enough to me :) 

Another strong reason to change is the lack of control that we have now with the GitHub repo (even Maujin can't make contributions freely because there is a requirement of another approval from other maintainers).

Still, we probably should keep GitHub as a backup at least (indefinitely).

**Comment by felipemanga** [comment]
I don't have any strong opinion regarding Github vs Gitlab, and I don't think many would be more interested in participating in a project just because of their choice of hosting. One of git's main features is its decentralized nature and migrating the code to a new remote (even if github were to suddenly disappear) is trivial. Having a mirror is just one more thing to maintain and possibly more confusing.

**Comment by maujin111** [comment]
For now just keep it here, GitHub it's not going to any place soon. Have the code in two places would be messy.

**Comment by Zughy** [comment]
Ok people, thanks for sharing your perspective. Closing then
